### PR TITLE
Support different Java versions garbage collection options

### PR DESCRIPTION
**Pull request by chonton**
### Type of change

_Select the type of your PR_

- Enhancement / new feature
- Refactoring

### Description

Choose garbage collection options based upon the version of java that is installed in the container.  Each image could potentially have different java versions.  This is a step towards supporting java 11 as well as java 8.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: docker-images/java-base/scripts/launch_java.sh [hunk 1]
```diff
@@ -16,6 +32,6 @@ export MALLOC_ARENA_MAX=2
 JAVA_OPTS="${JAVA_OPTS} -Dvertx.cacheDirBase=/tmp -Djava.security.egd=file:/dev/./urandom"
 
 # Enable GC logging for memory tracking
-JAVA_OPTS="${JAVA_OPTS} ${STRIMZI_GC_LOG_OPTS}"
+JAVA_OPTS="${JAVA_OPTS} $(get_gc_opts)"
```
Comment (tombentley) [inline docker-images/java-base/scripts/launch_java.sh]: Shouldn't we only do this when `STRIMZI_KAFKA_GC_LOG_ENABLED == true`?

Comment (scholzj) [reply docker-images/java-base/scripts/launch_java.sh]: I think it returns empty string when `STRIMZI_GC_LOG_ENABLED` is `false`, or?

What I wonder about is whether we really need two environment variables: `STRIMZI_KAFKA_GC_LOG_ENABLED` and `STRIMZI_GC_LOG_ENABLED`. But that is not really issue of this PR.

Comment (tombentley) [reply docker-images/java-base/scripts/launch_java.sh]: Oh yeah!

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 2]
```diff
@@ -94,15 +94,15 @@
...
+    public static final String ENV_VAR_STRIMZI_KAFKA_GC_LOG_ENABLED = "STRIMZI_KAFKA_GC_LOG_ENABLED";
+    public static final String ENV_VAR_STRIMZI_GC_LOG_ENABLED = "STRIMZI_GC_LOG_ENABLED";
 
     private static final String ANNO_STRIMZI_IO_DELETE_CLAIM = Annotations.STRIMZI_DOMAIN + "/delete-claim";
     @Deprecated
     private static final String ANNO_CO_STRIMZI_IO_DELETE_CLAIM = "cluster.operator.strimzi.io/delete-claim";
 
-    protected static final String DEFAULT_KAFKA_GC_LOGGING = "-verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCTimeStamps";
-    protected static final String DEFAULT_STRIMZI_GC_LOGGING = "-XX:NativeMemoryTracking=summary -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps";
+    protected static final String DEFAULT_KAFKA_GC_LOG_ENABLED = "true";
```
Comment (stanlyDoge) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: One nit. I would use prettier `String.valueOf(true)`.

Comment (chonton) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: done

**Comment by chonton** [comment]
No GC files in the pods.  (The KAFKA_GC_LOG_OPTS environment variable is not set in the image)

**Comment by scholzj** [comment]
Thanks for the PR @chonton

**Pull request merged**